plasmapy/diagnostics/thomson.py

```diff
@@ -0,0 +1,187 @@
+        Velocity of the fluid or electron component in the rest frame in
+        units convertable to cm/s
+
+    ion_vel : astropy.units.Quantity ndarray, shape [N,3]
+        Velocity vectors for each ion population relative to the
+        fluid velocit in units convertable to cm/s.
+
+    probe_n : float ndarray, shape [3]
+        Unit vector in the direction of the probe laser.
+
+    scatter_n : float ndarray, shape [3]
+        Unit vector pointing from the scattering volume to the detector.
+
+    Returns
+    -------
+    alpha : float
+        Mean scattering parameter. alpha > 1 corresponds to collective
+        scattering, while alpha < 1 indicates non-collective scattering.
+
+    Skw : astropy.units.Quantity ndarray
+        Computed spectral density function over the input wavelength range
+        with units of nm.
+
+    Notes
+    -----
+    For details, see "Plasma Scattering of Electromagnetic Radiation" by
+    Sheffield et al. ISBN 978-0123748775. This code is a modified version of
+    the program described therein.
+
+    For a concise summary of the relevant physics, see Chapter 5 of Derek
+    Schaeffer's thesis, DOI: 10.5281/zenodo.3766933
+
+    """
+
+    if np.sum(fract) != 1.0:
+        print("WARNING: Sum(fract) != 1.0. Normalizing array.")
+        fract = fract/np.sum(fract)
+
+    # Ensure unit vectors are normalized
+    probe_n = probe_n/np.linalg.norm(probe_n)
+    scatter_n = scatter_n/np.linalg.norm(scatter_n)
+
+    # Convert units
+    wavelength = wavelength.to(u.cm)
+    probe_wavelength = probe_wavelength.to(u.cm)
+    ne = ne.to(u.cm**-3)
+    Te = Te.to(u.eV)
+    Ti = Ti.to(u.eV)
+    ion_vel = ion_vel.to(u.cm/u.s)
+    fluid_vel = fluid_vel.to(u.cm/u.s)
+
+    # Define some constants
+    C = 2.99792458e10*u.cm/u.s  # speed of light
+    Me = 5.685e-16*u.eV/(u.cm/u.s)**2  # Electron mass in eV per (cm/s)^2
+    Mp = Me*1836.1  # Proton mass in same units
+    Mi = ion_mu*Mp  # mass of each ion species in same units
+    vTe = np.sqrt(Te/Me)  # Electron thermal velocity
+    vTi = np.sqrt(Ti/Mi)  # Ion thermal velocity
+    wpe = 5.64e4*(u.cm**1.5*u.rad/u.s)*np.sqrt(ne)
+
+    # Compute the ion velocity in the rest frame
+    ion_vel = fluid_vel + ion_vel
+
+    # Convert wavelengths to angular frequencies (electromagnetic waves, so
+    # phase speed is c)
+    ws = (2*np.pi*u.rad*C/wavelength).to(u.rad/u.s)
+    wl = (2*np.pi*u.rad*C/probe_wavelength).to(u.rad/u.s)
+
+    # Compute the frequency shift (required by energy conservation)
+    w = ws - wl
+
+    # Compute the wavenumbers in the plasma
+    # See Sheffield Sec. 1.8.1 and Eqs. 5.4.1 and 5.4.2
+    ks = np.sqrt(ws**2 - wpe**2)/C
+    kl = np.sqrt(wl**2 - wpe**2)/C
+
+    # Compute the wavenumber shift (required by momentum conservation)
+    sa = np.arccos(np.dot(probe_n, scatter_n))  # The scattering angle
+    k = np.sqrt(ks**2 + kl**2 - 2*ks*kl*np.cos(sa))  # Eq. 1.7.10 in Sheffield
```

Comment:
Comments like this one tend to be better as variable names:
```suggestion
    scattering_angle = np.arccos(np.dot(probe_n, scatter_n))
    k = np.sqrt(ks**2 + kl**2 - 2*ks*kl*np.cos(scattering_angle))  # Eq. 1.7.10 in Sheffield
```

And since this variable is used in a single expression, if you're worried about line length, I'm fine even with going full Unicode:
```suggestion
    scattering_φ = np.arccos(np.dot(probe_n, scatter_n))
    k = np.sqrt(ks**2 + kl**2 - 2*ks*kl*np.cos(scattering_φ))  # Eq. 1.7.10 in Sheffield
```